driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionAwsCredentialFromEnvironmentTest.java

```diff
@@ -192,6 +196,60 @@ public void shouldThrowMongoConfigurationIfSupplierReturnsDoesSomethingUnexpecte
         }
     }
 
+    @Test
+    public void shouldThrowMongoCryptIfNamedKmsProviderHasEmptyParametersInEncryptionSettings() {
+        assumeTrue(serverVersionAtLeast(4, 2));
+        assumeFalse(System.getenv().containsKey("AWS_ACCESS_KEY_ID"));
+        assumeTrue(isClientSideEncryptionTest());
+
+        Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>() {{
+            put("aws:name", new HashMap<>());
+        }};
+
+        Map<String, Supplier<Map<String, Object>>> kmsProviderPropertySuppliers = Mockito.mock(Map.class);
+        ClientEncryptionSettings settings = ClientEncryptionSettings.builder()
+                .keyVaultNamespace("test.datakeys")
+                .kmsProviders(kmsProviders)
+                .kmsProviderPropertySuppliers(kmsProviderPropertySuppliers)
+                .keyVaultMongoClientSettings(Fixture.getMongoClientSettings())
+                .build();
+
+        MongoCryptException e = assertThrows(MongoCryptException.class, () -> {
+            try (ClientEncryption ignore = createClientEncryption(settings)) {//NOP
+            }
+        });
+        assertTrue(e.getMessage().contains("On-demand credentials are not supported for named KMS providers."));
+        verifyNoInteractions(kmsProviderPropertySuppliers);
```

Comment:
[optional]

It seems that there should be no interactions with the suppliers, while the interactions with `kmsProviderPropertySuppliers` itself are not conceptually prohibited. I understand, that the current implementation happens to avoid any interactions with `kmsProviderPropertySuppliers`, but it would have been better to make the assertions less depending on the implementation detail.

The same applies to `shouldThrowMongoCryptIfNamedKmsProviderHasEmptyParametersInAutoSettings`.

Reply:
I have adjusted the test behavior to verify that only `Supplier.get` has not been invoked. This is done using the following setup:

```
 Map<String, Supplier<Map<String, Object>>> kmsProviderPropertySuppliers = new HashMap<>();
        kmsProviderPropertySuppliers.put("aws:name", () -> Assertions.fail("Supplier should not be called"));
```